src/main/java/com/example/demo/repository/impl/BookRepositoryImpl.java

```diff
@@ -2,51 +2,49 @@
 
 import com.example.demo.model.Book;
 import com.example.demo.repository.BookRepository;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
 import java.util.List;
-import org.hibernate.Session;
-import org.hibernate.SessionFactory;
-import org.hibernate.Transaction;
-import org.springframework.beans.factory.annotation.Autowired;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Repository;
 
+@RequiredArgsConstructor
 @Repository
 public class BookRepositoryImpl implements BookRepository {
-    @Autowired
-    private final SessionFactory sessionFactory;
+    private final EntityManagerFactory entityManagerFactory;
 
-    public BookRepositoryImpl(SessionFactory sessionFactory) {
-        this.sessionFactory = sessionFactory;
+    @Override
+    public List<Book> getAll() {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            return entityManager.createQuery(
+                    "SELECT u FROM Book u", Book.class).getResultList();
+        }
     }
 
     @Override
-    public Book save(Book book) {
-        Session session = null;
-        Transaction transaction = null;
-        try {
-            session = sessionFactory.openSession();
-            transaction = session.beginTransaction();
-            session.save(book);
-            transaction.commit();
-            return book;
-        } catch (Exception e) {
-            if (transaction != null) {
-                transaction.rollback();
-            }
-            throw new RuntimeException("Can't save book into DB " + book);
-        } finally {
-            if (session != null) {
-                session.close();
-            }
+    public Optional<Book> findBookById(Long id) {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            Book book = entityManager.find(Book.class, id);
+            return Optional.ofNullable(book);
         }
     }
 
     @Override
-    public List<Book> findAll() {
-        try (Session session = sessionFactory.openSession()) {
-            return session.createQuery(
-                    "SELECT u FROM Book u", Book.class).getResultList();
-        } catch (Exception e) {
-            throw new RuntimeException("Can't get all books from DB ", e);
+    public Book createBook(Book book) {
+        EntityTransaction transaction = null;
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
```

Comment:
don't use try with resources and transactions, session will close before catch block